src/io/flatgeobuf/object_store_reader.rs

```diff
@@ -0,0 +1,43 @@
+use async_trait::async_trait;
+use bytes::Bytes;
+use http_range_client::{AsyncHttpRangeClient, Result as HTTPRangeClientResult};
+use object_store::path::Path;
+use object_store::ObjectStore;
+
+pub struct ObjectStoreWrapper<T: ObjectStore> {
+    pub location: Path,
+    pub reader: T,
+}
+
+#[async_trait]
+impl<T: ObjectStore> AsyncHttpRangeClient for ObjectStoreWrapper<T> {
+    /// Send a GET range request
+    async fn get_range(&self, _url: &str, range: &str) -> HTTPRangeClientResult<Bytes> {
+        let split_range = range.split('-').collect::<Vec<_>>();
+        let start_range = split_range[0].parse::<usize>().unwrap();
+        let end_range = split_range[1].parse::<usize>().unwrap();
```

Comment:
`range` (at least in my testing) looks like `bytes=0-12943` - so your split is insufficient.

I've done `let split_range = range.split(&['=', '-']).collect::<Vec<_>>();` and then incremented the indexes on line 17, 18.

Still looking, but I'm getting a response with this change at least :rocket:

Reply:
A refactor including errors:

```rust
        let bytes_range = range
            .split(&['=', '-'])
            .filter_map(|n| n.parse::<usize>().ok());
        if let Some((start_range, end_range)) = bytes_range.collect_tuple() {
            // Add one to the range because HTTP range strings are end-inclusive
            let bytes = self
                .reader
                .get_range(&self.location, start_range..end_range + 1)
                .await
                .map_err(|_| HttpError::HttpStatus(500))?;
            return Ok(bytes);
        }
        Err(HttpError::HttpStatus(500))
```

Reply:
```rust
use itertools:Itertools; #collect_tuple
use http_range_client::HttpError;
```